course_dashboard/stats.py

```diff
@@ -175,3 +175,39 @@ def total(self):
     def daily_average(self):
         """Average enrollments per day"""
         return self.total() * 1. / self.day_span()
+
+from certificates.models import GeneratedCertificate, CertificateStatuses
+
+class CertificateStats(object):
+    """Provide certificate stats for a given course."""
+
+    def __init__(self, course_key):
+        self.course_key = course_key
+        self.certificates = GeneratedCertificate.objects.filter(course_id=course_key)
+
+    @property
+    def notpassing(self):
+        """Return the number of failed certificates"""
+        notpassing = 0
+        if not self.certificates:
+            return None
+        for certificate in self.certificates:
```

Comment:
Nécessite d'itérer sur tous les certificats, alors que `GeneratedCertificate.objects.filter(course_id=course_key).filter(status==CertificateStatuses.notpassing).count()` fait la même chose.